### PR TITLE
native plumbing: setup tamagui, pull in components from POC

### PR DESCRIPTION
**Pull request by patosullivan**
Pulls in most of the work from #3348 around adding tamagui to the app and getting it to work cross-platform.

Does *not* pull in the tamagui components to the web app (since that app still needs to follow normal production schedule releases), but I did temporarily pull them in to verify they work.

Also fixes module resolution issues and the vite issues from that branch.

Fixing the ESM issue with @tamagui/vite-plugin required creating a patch based on this branch that got merged to their repo yesterday: https://github.com/tamagui/tamagui/commit/130bcf52295f0b1db55ea7fdbcbe78ffffe6b48e

Patch should only be temporary, tamagui releases pretty frequently.

*Note*: Any calls to `db` methods were commented out pending Dan's work to get the db/api side up to date.

fixes LAND-1727

### CODE REVIEW

File: apps/tlon-mobile/src/components/GroupListItem.tsx [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  const handlePress = useCallback(() => {
+    onPress?.(model);
+  }, [onPress, model]);
+  const handleLongPress = useCallback(() => {
+    onLongPress?.(model);
+  }, [onLongPress, model]);
+  const unreadCount = useMemo(() => {
+    return 0;
+    // return db.getUnreadChannelCount(model);
+  }, [model]);
```
[the remark this answers is not in the document]
Comment (dnbrwstr) [reply apps/tlon-mobile/src/components/GroupListItem.tsx:23]: (Actually, once we move this to `ui`, we should remove the db query entirely).

File: apps/tlon-mobile/package.json [hunk 2]
```diff
@@ -40,8 +40,10 @@
     "@react-navigation/native-stack": "^6.9.13",
     "@realm/react": "^0.6.2",
     "@tloncorp/shared": "*",
+    "@tloncorp/ui": "*",
     "@urbit/aura": "^1.0.0",
     "@urbit/http-api": "^3.1.0-dev-2",
+    "@urbit/sigil-js": "^2.2.0",
```
[the remark this answers is not in the document]
Comment (patosullivan) [reply apps/tlon-mobile/package.json:46]: Looks like no, will remove.

File: apps/tlon-web/package.json [hunk 3]
```diff
@@ -56,6 +56,7 @@
     "@radix-ui/react-toggle": "^1.0.2",
     "@radix-ui/react-toggle-group": "^1.0.4",
     "@radix-ui/react-tooltip": "^1.0.0",
+    "@tamagui/vite-plugin": "1.92.1",
```
[the remark this answers is not in the document]
Comment (patosullivan) [reply apps/tlon-web/package.json:59]: Updated other tamagui modules to 1.92.1, seems to work (with a few minor changes).

File: packages/ui/package.json [hunk 4]
```diff
@@ -7,11 +7,23 @@
...
+  },
   "dependencies": {
-    "@tamagui/animations-moti": "^1.89.26",
-    "@tamagui/get-token": "^1.89.26",
-    "@urbit/sigil-js": "^2.2.0",
+    "@tamagui/animations-moti": "1.89.26",
+    "@tamagui/get-token": "1.89.26",
+    "@tamagui/image": "1.89.26",
+    "@uidotdev/usehooks": "^2.4.1",
+    "react": "^18.2.0",
```
[the remark this answers is not in the document]
Comment (patosullivan) [reply packages/ui/package.json:19]: Appears to work as a peer dependency, so I'll switch it over.

File: packages/ui/src/components/Avatar.tsx [hunk 5]
```diff
@@ -1,13 +1,24 @@
...
 
 export function Avatar({
   contact,
   ...props
 }: {
   contact: Client.Contact;
 } & ViewProps) {
+  // on native, we have to pass height/width for the source prop, on web we want to use other attributes
+  // to set those
+  const nativeDims = useMemo(
```
[the remark this answers is not in the document]
Comment (patosullivan) [reply packages/ui/src/components/Avatar.tsx:14]: I didn't work on this piece, this is pulled directly from the Hybrid POC branch. Might need to ask @latter-bolden about this one.

File: packages/ui/src/components/GroupList.tsx [hunk 6]
```diff
@@ -0,0 +1,66 @@
+import { SectionList, StyleProp, ViewStyle } from "react-native";
```
[the remark this answers is not in the document]
Comment (patosullivan) [reply packages/ui/src/components/GroupList.tsx:1]: I rewrote the component using tamagui's YGroup component, looks/works fine with no react-native dependencies.

File: packages/ui/src/index.ts [hunk 7]
```diff
@@ -1,7 +1,16 @@
-export * from "./components/Button";
-export * from "./components/TamaguiProvider";
-export * from "./tamagui.config";
+// export { Circle, useTheme, useStyle, View, ZStack } from "tamagui";
+export * from "./core";
```
[the remark this answers is not in the document]
Comment (dnbrwstr) [reply packages/ui/src/index.ts:1]:
```suggestion
export * from "./core";
```

File: packages/ui/src/components/Avatar.tsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (patosullivan) [reply packages/ui/src/components/Avatar.tsx:14]: Added TODO.

File: packages/ui/src/components/GroupListItem/ListItemContent.tsx [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+import { ListItem, type ListItemProps } from "../ListItem";
+import { useMemo } from "react";
+
+export default function ListItemContent({
+  model,
+  onPress,
+  onLongPress,
+  ...props
+}: ListItemProps<Client.Group>) {
+  const unreadCount = useMemo(() => {
```
[the remark this answers is not in the document]
Comment (patosullivan) [reply packages/ui/src/components/GroupListItem/ListItemContent.tsx]: Made it a prop in one of the latest commits.

File: packages/ui/src/components/GroupListItem/index.web.tsx [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+import { Icon } from "../Icon";
+import ListItemContent from "./ListItemContent";
+
+export const GroupListItem = ({
+  model,
+  onPress,
+  onLongPress,
+  ...props
+}: ListItemProps<Client.Group>) => {
+  const attributes = useLongPress(
```
[the remark this answers is not in the document]
Comment (patosullivan) [reply packages/ui/src/components/GroupListItem/index.web.tsx:13]: Added TODO.

File: packages/ui/src/components/GroupOptionsSheet.tsx [hunk 10]
```diff
@@ -0,0 +1,148 @@
...
+      <Sheet.Frame>
+        <Sheet.Handle paddingTop="$xl" />
+        <View
+          gap="$xl"
+          paddingHorizontal="$2xl"
+          paddingTop="$xl"
+          paddingBottom="$4xl"
+        >
+          <Stack paddingBottom="$m" flexDirection="column">
+            <Text fontSize="$l" fontWeight="500">
```
[the remark this answers is not in the document]
Comment (patosullivan) [reply packages/ui/src/components/GroupOptionsSheet.tsx:31]: Added TODO

File: packages/ui/src/components/UrbitSigil.native.tsx [hunk 11]
```diff
@@ -0,0 +1,39 @@
+import sigil from "@urbit/sigil-js/dist/core";
```
[the remark this answers is not in the document]
Comment (patosullivan) [reply packages/ui/src/components/UrbitSigil.native.tsx:1]: Added a comment.

File: packages/ui/src/components/UrbitSigil.native.tsx [hunk 12]
```diff
@@ -0,0 +1,39 @@
...
+  const theme = useTheme();
+  const sigilXml = useMemo(
+    () =>
+      validShip
+        ? sigil({
+            point: ship,
+            detail: "none",
+            size: 12,
+            space: "none",
+            foreground: "#ffffff",
```
[the remark this answers is not in the document]
Comment (patosullivan) [reply packages/ui/src/components/UrbitSigil.native.tsx:19]: Added TODO

File: packages/ui/src/components/UrbitSigil.native.tsx [hunk 13]
```diff
@@ -0,0 +1,39 @@
...
+            background: theme.darkBackground.val,
+          })
+        : null,
+    [ship]
+  );
+  return (
+    <View
+      ref={ref}
+      width={20}
```
[the remark this answers is not in the document]
Comment (patosullivan) [reply packages/ui/src/components/UrbitSigil.native.tsx:28]: Added TODO

File: packages/ui/src/components/UrbitSigil.tsx [hunk 14]
```diff
@@ -1,6 +1,5 @@
-import sigil from "@urbit/sigil-js/dist/core";
+import sigil from "@urbit/sigil-js";
```
[the remark this answers is not in the document]
Comment (patosullivan) [reply packages/ui/src/components/UrbitSigil.tsx]: Based on @alecananian's comment it sounds like we need to leave this as-is. Added a note to this component as well to explain.

File: packages/ui/src/components/ListItem.tsx [hunk 15]
```diff
@@ -0,0 +1,240 @@
...
+
+const ListItemMainContent = styled(YStack, {
+  flex: 1,
+  justifyContent: "center",
+  height: "$4xl",
+  paddingVertical: "$xs",
+});
+
+const ListItemTitle = styled(SizableText, {
+  // lineHeight: 0,
```
[the remark this answers is not in the document]
Comment (patosullivan) [reply packages/ui/src/components/ListItem.tsx]: Removed.

**Review by dnbrwstr — APPROVED** [review]
Great, LGTM!

**Pull request merged**